wetterdienst/provider/noaa/ghcn/api.py

```diff
@@ -0,0 +1,171 @@
+from wetterdienst.metadata.timezone import Timezone
+from wetterdienst.provider.noaa.ghcn.parameter import NoaaGhcnParameter
+from wetterdienst.provider.noaa.ghcn.unit import NoaaGhcnUnit
+
+
+class NoaaGhcnDatasetBase(Enum):
+    DAILY = "daily"
+
+
+class NoaaGhcnResolution(Enum):
+    DAILY = Resolution.DAILY.value
+
+
+class NoaaGhcnValues(ScalarValuesCore):
+    _string_parameters = None
+
+    _integer_parameters = None
+
+    _irregular_parameters = None
+
+    _data_tz = Timezone.USA
+
+    _base_url = (
+        "https://www1.ncdc.noaa.gov/pub/data/ghcn/daily/by_station/{station_id}.csv.gz"
+    )
+
+    def _tidy_up_df(self, df: pd.DataFrame, dataset) -> pd.DataFrame:
+        pass
+
+    def _collect_station_parameter(
+        self, station_id: str, parameter, dataset
+    ) -> pd.DataFrame:
+        """
+        Collection method for NOAA GHCN data. Parameter and dataset can be ignored as data
+        is provided as a whole.
+
+        :param station_id: station id of the station being queried
+        :param parameter: parameter being queried
+        :param dataset: dataset being queried
+        :return: dataframe with read data
+        """
+        # parameter and dataset can be ignored as data is provided as a whole
+
+        url = self._base_url.format(station_id=station_id)
+
+        r = requests.get(url)
+
+        r.raise_for_status()
+
+        file = BytesIO(r.content)
+
+        columns = (
+            Columns.STATION_ID.value,
+            Columns.DATE.value,
+            Columns.PARAMETER.value,
+            Columns.VALUE.value,
+            Columns.QUALITY.value,
+        )
+
+        df = pd.read_csv(file, sep=",", compression="gzip", header=None)
+
+        df = df.iloc[:, :-3]
+
+        df.columns = columns
+
+        df.loc[:, Columns.PARAMETER.value] = df.loc[
+            :, Columns.PARAMETER.value
+        ].str.lower()
+
+        return df
+
+
+class NoaaGhcnRequest(ScalarRequestCore):
+    _dataset_base = NoaaGhcnDatasetBase
+    _parameter_base = NoaaGhcnParameter
+    _dataset_tree = NoaaGhcnParameter
+
+    _base_url = "https://www1.ncdc.noaa.gov/pub/data/ghcn/daily/ghcnd-stations.txt"
+
+    _data_range = DataRange.FIXED
+
+    _has_datasets = True
+    _unique_dataset = True
+
+    _period_type = PeriodType.FIXED
+    _period_base = Period.HISTORICAL
+
+    _resolution_type = ResolutionType.FIXED
+    _resolution_base = NoaaGhcnResolution
+
+    provider = Provider.NOAA
+    kind = Kind.OBSERVATION
+
+    _unit_tree = NoaaGhcnUnit
+
+    _values = NoaaGhcnValues
+
+    _tz = Timezone.USA
+
+    def __init__(
+        self,
+        parameter: List[str],
+        start_date: Optional[Union[str, datetime, pd.Timestamp]] = None,
+        end_date: Optional[Union[str, datetime, pd.Timestamp]] = None,
+        humanize: bool = True,
+        tidy: bool = True,
+        si_units: bool = True,
+    ):
+        super(NoaaGhcnRequest, self).__init__(
```

Comment:
This looks like you are trying to make a dependency injection. For this purpose you can define a `@classmethod`  `make`

I am having a tiny example for you: 
```
class DenseNeuralNetworkRegressor:
    """
    A fully connected feed forward neural network regressor. Applies normalization to the input data.
    """
    def __init__(
            self,
            hidden_layer_sizes: List[int],
            output_dimension: int = 1
    ):
        hidden_layers = [Dense(units=units) for units in hidden_layer_sizes]
        self.model = Sequential(
            [
                Normalization(name='normalization'),
                *hidden_layers,
                Dense(units=output_dimension)
            ]
        )

    @classmethod
    def make(
            cls,
            hidden_layer_sizes: List[int],
            optimizer: Optimizer,
            loss: Loss = MeanSquaredError()
    ) -> 'DenseNeuralNetworkRegressor':
        """
        Constructs a default DenseNeuralNetworkRegressor with the specified hidden layers and optimizer.

        Args:
            hidden_layer_sizes: List of numbers of units for the hidden layers.
            optimizer: Optimizer to use for training.
            loss: Loss to use for training.

        Returns:
            A compiled tf.keras.Model.
        """
        dnn_regressor = DenseNeuralNetworkRegressor(
            hidden_layer_sizes=hidden_layer_sizes
        )
        dnn_regressor.model.compile(
            optimizer=optimizer,
            loss=loss
        )
        return dnn_regressor
``` 
This will make it much easier to test btw.

Reply:
Dear @meteoDaniel ,

this is just a regular super call with no further intentions. Hoever thanks for the hint. 

By saying it is much easier to test you mean for the sake of mocking?